manager/src/main/java/com/redhat/service/bridge/manager/workers/id/WorkerIdProvider.java

```diff
@@ -4,13 +4,33 @@
 
 import javax.enterprise.inject.Produces;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class WorkerIdProvider {
 
-    private static final String WORKER_ID = UUID.randomUUID().toString();
+    private static final String RANDOM_STATIC_WORKER_ID = UUID.randomUUID().toString();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkerIdProvider.class);
+
+    private String findPodNameFromKubernetes() {
+        String hostname = System.getenv("HOSTNAME");
+        LOGGER.trace("POD id from hostname: {}", hostname);
+
+        if (hostname != null && !hostname.isEmpty() && !"localhost".equals(hostname)) {
+            return "worker-" + hostname;
+        } else {
+            return null;
+        }
+    }
 
     @Produces
     public String getWorkerId() {
-        return WORKER_ID;
+        String podNameFromKubernetes = findPodNameFromKubernetes();
+        if (podNameFromKubernetes == null) {
+            return RANDOM_STATIC_WORKER_ID;
```

Comment:
maybe it would be good to log this. 
Just an headsup: @manstis I think it would be great if in your logging PR we add the identifier of the manager pod/instance in all the logs. Not something we have to include in your PR, just saying let's consider it in another ticket

Reply:
Great idea @r00ta.. I've been trying to think of more data for the base message (sub-classes can have more.. but my PR is just the beginning and something _concrete_ to discuss vs trying to discuss something hypothetically) :+1:

Reply:
@r00ta I'll wait for this PR to be merged as I need to move `WorkerIdProvider` out of `manager` and into somewhere more accessible (less dependent). Perhaps `infra` would be a good home for it.